tools/ete/ete_lineage_generator.xml

```diff
@@ -12,9 +12,7 @@ python '$__tool_directory__/ete_lineage_generator.py'
 #if $ranks.levels == "full"
     -f
 #elif $ranks.levels == "manual"
-    #for $rank in $ranks.manualranks:
```

Comment:
A simpler fix for the bug is to only change this line to:
```
    #for $rank in str($ranks.manualranks).split(','):
```
and restore the Python code.

Also, you should add `optional="false"` to the `manualranks` param, there's no point of running the tool with no rank, right?

Reply:
done